src/ToDo/Business/TaskService.cs

```diff
@@ -40,4 +40,16 @@ public async Task DeleteAsync(ToDoTask task, CancellationToken ct)
 
 		_messenger.Send(new EntityMessage<ToDoTask>(EntityChange.Delete, task), task.ListId);
 	}
+
+	/// <inheritdoc />
+	public async ValueTask<IImmutableList<ToDoTask>> GetAllAsync(string displayName = "", CancellationToken ct = default)
+	{
+		Task<TaskReponseData<TaskData>> getMethod = string.IsNullOrWhiteSpace(displayName) ? _client.GetAllAsync(ct) : _client.GetByFilterAsync(displayName, ct);
+		return ((await getMethod).Value ?? Enumerable.Empty<TaskData>())
+		.Where(data => data.ParentList?.Id is not null)
+		.Select(data =>
+		{
+			return new ToDoTask(data.ParentList?.Id ?? string.Empty, data);
+		}).ToImmutableList();
```

Comment:
Just a recommendation for next time: As on the line 49 you checked for `data.ParentList?.Id is not null`, it's  perfectly legit to "forgive the null" here:

```suggestion
		.Where(data => data.ParentList?.Id is not null)
		.Select(data => new ToDoTask(data.ParentList!.Id!, data))
		.ToImmutableList();
```

It's for those cases where the compiler cannot detect it by itself (yet) that the `!` exists.